src/platform_impl/windows/icon.rs

```diff
@@ -160,3 +163,93 @@ pub fn unset_for_window(hwnd: HWND, icon_type: IconType) {
         SendMessageW(hwnd, WM_SETICON, icon_type as usize, 0);
     }
 }
+
+#[derive(Debug)]
+struct RaiiCursor {
+    handle: HCURSOR,
+}
+
+#[derive(Clone, Debug)]
+pub struct WinCursor {
+    inner: Arc<RaiiCursor>,
+}
```

Comment:
You don't have to nest like that, could just have a single `drop` which checks the ref counting stuff.

Reply:
True, unneeded indirection that was copied from `WinIcon`, but I simplified it anyway.